drivers/usb/device/usb_dc_stm32.c

```diff
@@ -89,6 +85,18 @@ static const struct gpio_dt_spec ulpi_reset =
 	GPIO_DT_SPEC_GET_OR(DT_PHANDLE(DT_INST(0, st_stm32_otghs), phys), reset_gpios, {0});
 #endif
 
+/**
+ * The following defines are used to map the value of the "maxiumum-speed"
+ * DT property to the corresponding definition used by the STM32 HAL.
+ */
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHYC) || \
+	defined(USB_OTG_HS_EMB_PHY)
```

Comment:
This breaks high-speed usb on various chips as `USB_OTG_HS_EMB_PHYC` and `USB_OTG_HS_EMB_PHY` is always defined.

Reply:
shouldn't it be something like this:
```diff
@@ -91,8 +91,8 @@ static const struct gpio_dt_spec ulpi_reset =
  * The following defines are used to map the value of the "maxiumum-speed"
  * DT property to the corresponding definition used by the STM32 HAL.
  */
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHYC) || \
-	defined(USB_OTG_HS_EMB_PHY)
+#if DT_ENUM_IDX(DT_DRV_INST(0), maximum_speed) == 1 && \
+	(defined(CONFIG_SOC_SERIES_STM32H7X) || USB_OTG_HS_EMB_PHYC || USB_OTG_HS_EMB_PHY)
 #define USB_DC_STM32_HIGH_SPEED             USB_OTG_SPEED_HIGH_IN_FULL
 #else
 #define USB_DC_STM32_HIGH_SPEED             USB_OTG_SPEED_HIGH
```

Reply:
Would you mind raising an issue with an example of board reference on which issue could be seen ?

Reply:
done, #89831